crates/egui/src/text_selection/text_cursor_state.rs

```diff
@@ -291,6 +292,18 @@ pub fn is_word_char(c: char) -> bool {
     c.is_ascii_alphanumeric() || c == '_'
 }
 
+/// Determines whether a character is a component of a string.
+///
+/// Punctuation marks that are commonly used with strings are considered string components.
+pub fn is_string_component(c: char) -> bool {
+    if c == '_' || c == '-' || c == ':' || c == '/' || c == '.' || c == '\\' || c == '@' || c == '#'
```

Comment:
You can use `matches!` instead. See <https://rust-lang.github.io/rust-clippy/master/index.html#/string_lit_chars_any>.

Reply:
I don't think it's a place that requires high speed.

I think In general, rust is fast enough, so it's better to think about readability rather than speed.

Reply:
`matches!(c, '_' | '-' | ':' | …)` is much more readable than repeating `c ==` over and over again

Reply:
> `matches!(c, '_' | '-' | ':' | …)` is much more readable than repeating `c ==` over and over again

Okay. Edit has been completed.

in URL, '?' and '=' were also added because they are often used.